src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
I agree. Exports should be transitive. Sorry. I think I wasn't clear enough in that last comment.